File: submissions/4220-minimum-operations-to-sort-a-string/solution.java
import java.util.*;

class Solution {
    public int minOperations(String s) {
        int n = s.length();
        String act = s;
        
        char[] arr = s.toCharArray();
        Arrays.sort(arr);
        s = new String(arr);

        if(s.equals(act)) return 0;
        if(s.length() < 3) return -1;

        String sub1 = act.substring(0, n - 1);
        System.out.println(sub1);

        char[] sub1Arr = sub1.toCharArray();
        Arrays.sort(sub1Arr);
        sub1 = new String(sub1Arr);
        System.out.println(sub1);

        String sub2 = act.substring(1, n);
        char[] sub2Arr = sub2.toCharArray();
        Arrays.sort(sub2Arr);
        sub2 = new String(sub2Arr);

        String reverse1 = new StringBuilder(act).reverse().toString();

        System.out.println(s);
        System.out.println(sub1 + act.substring(n - 1, n));

        if((sub1 + act.substring(n - 1, n)).equals(s) || (act.substring(0, 1) + sub2).equals(s)){
            return 1;
        } 
        else {
            sub1 += act.substring(n - 1, n);

            String nestedsub1 = sub1.substring(1);
            char[] nested1Arr = nestedsub1.toCharArray();
            Arrays.sort(nested1Arr);
            nestedsub1 = new String(nested1Arr);

            sub2 = act.substring(0, 1) + sub2;

            String nestedsub2 = sub2.substring(0, sub2.length() - 1);
            char[] nested2Arr = nestedsub2.toCharArray();
            Arrays.sort(nested2Arr);
            nestedsub2 = new String(nested2Arr);

            if((sub1.substring(0,1) + nestedsub1).equals(s) || 
               (nestedsub2 + sub2.substring(sub2.length()-1)).equals(s))
                return 2;

            return 3;
        }
    }
}

File: submissions/4220-minimum-operations-to-sort-a-string/solution.cpp
class Solution {
public:
    int minOperations(string s) {
        int n = s.size();
        string act = s;
        sort(s.begin(),s.end());
        if(s == act) return 0;
        if(s.size() < 3) return -1;
        string sub1 = act.substr(0,n-1);
        cout<<sub1<<endl;
        sort(sub1.begin(),sub1.end());
        cout<<sub1<<endl;
        string sub2 = act.substr(1,n);
        sort(sub2.begin(),sub2.end());
        string reverse1 = act;
        cout<<s<<endl;
        cout<<sub1 + act.substr(n-1,n);
        reverse(reverse1.begin(),reverse1.end());
        if(sub1 + act.substr(n-1,n) == s || act.substr(0,1) + sub2 == s ){
            return 1;
        }
        
        // else if(reverse1 == s){
        //     return 3;
        // }
        else{
            sub1 += act.substr(n-1,n);
            string nestedsub1 = sub1.substr(1,sub1.size());
            sort(nestedsub1.begin(),nestedsub1.end());
            sub2 = act.substr(0,1) + sub2;
            string nestedsub2 = sub2.substr(0,sub2.size()-1);
            sort(nestedsub2.begin(),nestedsub2.end());
            if(sub1.substr(0,1) + nestedsub1 == s || nestedsub2 + sub2.substr(sub2.size()-1,sub2.size()) == s) return 2;
            return 3;
        }
    }
};
